tests/e2e/performance/test_pvc_multi_clone_performance.py

```diff
@@ -0,0 +1,408 @@
+import datetime
+import logging
+import os
+import subprocess
+import tempfile
+import time
+
+import yaml
+
+from ocs_ci.framework import config
+from ocs_ci.framework.testlib import (
+    skipif_ocs_version,
+    skipif_ocp_version,
+    E2ETest,
+    performance,
+)
+from ocs_ci.ocs import constants, exceptions
+from ocs_ci.ocs.cluster import CephCluster
+from ocs_ci.ocs.resources import pod
+from ocs_ci.utility.utils import ocsci_log_path
+
+log = logging.getLogger(__name__)
+
+
+@performance
+@skipif_ocp_version("<4.6")
+@skipif_ocs_version("<4.6")
+class TestPvcMultiClonePerformance(E2ETest):
+    """
+    Tests to measure PVC clones creation performance ( time and speed)
+    The test is supposed to create the maximum number of clones for one PVC
+    """
+
+    def test_pvc_multiple_clone_performance(
+        self,
+        interface_iterate,
+        teardown_factory,
+        storageclass_factory,
+        pvc_factory,
+        pod_factory,
+    ):
+        """
+        1. Creating PVC
+           PVS size is calculated in the test and depends on the storage capacity, but not less then 1 GiB
+           it will use ~75% capacity of the Storage, Min storage capacity 1 TiB
+        2. Fill the PVC with 70% of data
+        3. Take a clone of the PVC and measure time and speed of creation by reading start creation and end creation
+            times from relevant logs
+        4. Repeat the previous step number of times (maximal num_of_clones is 512)
+        5. Rrint all measured statistics for all the clones.
+
+        Raises:
+            StorageNotSufficientException: in case of not enough capacity on the cluster
+
+        """
+        num_of_clones = 512
+
+        # Getting the total Storage capacity
+        ceph_cluster = CephCluster()
+        ceph_capacity = int(ceph_cluster.get_ceph_capacity())
+
+        # Use 70% of the storage capacity in the test
+        capacity_to_use = int(ceph_capacity * 0.7)
+
+        # since we do not want to use more then 65%, we add 35% to the needed
+        # capacity, and minimum PVC size is 1 GiB
+        need_capacity = int((num_of_clones + 2) * 1.35)
```

Comment:
I would suggest to change the logic to something generic, In future we might need to revisit this TC to increase the num_of_clone count, which can lead to need_capacity value higher than capacity_to_use.

Reply:
@ramkiperiy  - I don't really understand what do you mean by generic logic.Currently we suppose that each 
clone needs at least 1GB and test that need_capactiy (which is number_of_clones * 1.35)  in GB is smaller that capacity_to_use. If we increase in the future num_of_clones count, this really may lead to the situation when we have no potential space to create all clones and we will check this in the beginning of the test and exit if we have no enough space. Sounds reasonable enough

Reply:
> I would suggest to change the logic to something generic, In future we might need to revisit this TC to increase the num_of_clone count, which can lead to need_capacity value higher than capacity_to_use.

IMO, we can not used "very small" system for large number of clones, so, i do not see any problem in this logic.